### PR TITLE
Add optional figure captions for image alt text

### PR DESCRIPTION
**Pull request by sevmonster**
Adds new config variable `img_alt_text_use_figure` that enables a new template that wraps `img`s with `alt` text inside `figures` with `figcaption`s. The styling was inspired by [obsidian-image-captions](https://github.com/alangrainger/obsidian-image-captions/blob/main/styles.css).

The config value is disabled by default so this PR should not introduce any changes in the default configuration.

Also took the liberty of cleaning up some formatting while around.

image

### CODE REVIEW

**Comment by dwrolvink** [comment]
Very nice! 

Accepted the change and made it enabled by default, because I assume that people that take the effort to fill in the alt text do this mostly because they have a plugin that shows it, and very seldomly because they are worried about blind people / having the alt text showing if the image decides to get lost.

Small note: I do like cleaning up the code, in fact I've just picked up pylint and ruff yesterday and started cleaning up too by coincidence. But mixing in removing spaces with actual functional changes makes it hard to parse the changes of the new feature. So where possible do that in seperate commits. Not a big issue just thought I'd mention it.

**Comment by dwrolvink** [comment]
Decided that a centered image looks better, so I changed it to center the image. Did you prefer left-aligned, or did you just not test it with small images?

**Comment by sevmonster** [comment]
I also considered that (and I did think it would look better) but seeing as how images are not centered by default, I thought it safer to go with the flow. If people don't like it they can always override it I suppose.
